cl/recap/tasks.py

```diff
@@ -897,47 +838,49 @@ def process_case_query_page(self, pk):
             content_updated = True
 
     if pq.debug:
-        mark_pq_successful(pq, d_id=d.pk)
-        self.request.chain = None
+        await mark_pq_successful(pq, d_id=d.pk)
         return {"docket_pk": d.pk, "content_updated": False}
 
-    try:
-        d.save()
-        add_bankruptcy_data_to_docket(d, data)
-    except IntegrityError as exc:
-        logger.warning(
-            "Race condition experienced while attempting docket save."
-        )
-        error_message = "Unable to save docket due to IntegrityError."
-        if self.request.retries == self.max_retries:
-            mark_pq_status(pq, error_message, PROCESSING_STATUS.FAILED)
-            self.request.chain = None
+    retries = 5
+    while True:
```

Comment:
This while loop retry thing is repetitive. Is there a better way to factor it out?

Reply:
I didn't see an obvious better option here, this can always be refactored later as needed.

Reply:
I thought you might have an inspiring idea here. The only thing I can think of is to put this kind of code into a decorated function, but that's really not fun.